cellxgene_schema_cli/cellxgene_schema/validate.py

```diff
@@ -737,19 +737,15 @@ def _validate_dataframe(self, df_name: str):
 
     def _validate_colors_in_uns_dict(self, uns_dict: dict) -> None:
         df = getattr_anndata(self.adata, "obs")
-        df_definition = self._get_component_def("obs")
 
         # Mapping from obs column name to number of unique categorical values
         category_mapping = {}
 
-        if "columns" in df_definition:
-            for column_name, column_def in df_definition["columns"].items():
-                if column_name not in df.columns:
-                    # Skip this, dataframe validation should already append an error for this
-                    continue
-
-                if column_def.get("type") == "categorical":
-                    category_mapping[column_name] = df[column_name].nunique()
+        # Check for categorical dtypes in the dataframe directly
+        for column_name in df.columns:
+            column = df[column_name]
+            if column.dtype == "category":
```

Comment:
nit: use `column.dtype.name` for comparison, as this comparison has weird edge cases that throw errors instead of failing gracefully in older versions of numpy (source: https://stackoverflow.com/questions/26924904/check-if-dataframe-column-is-categorical#comment123655621_26925340)

Reply:
```suggestion
            if column.dtype.name == "category":
```

Reply:
makes sense - i also updated the piece of code i copied and pasted this snippet from 😅